guides/release/templates/handlebars-basics.md

```diff
@@ -1,85 +1,203 @@
+
+For example, either of these commands will generate `.hbs` Template files (and other things!) in your app:
+
+```sh
+ember generate component my-component-name
+ember generate route my-route-name
+```
+
+## Template restrictions
+
+A typical, modern web app is made of dozens of files that have to all be combined together into something the browser can understand. Ember does this work for you with zero configuration, but as a result, there are some rules to follow when it comes to adding assets into your HTML.
+
+You cannot use script tags directly within a template, and should use [actions]() or [Route Lifecycle Hooks]() to make your app responsive to user interactions and new data. If you are working with a non-Ember JavaScript library and need to use a `js` file from it, see the Guide section [Addons and Dependencies]().
+
+Similarly, you should not add links to CSS Stylesheets within the `hbs` file. Style rules should go in the `app/styles` directory instead. `app/styles/app.css` is included in your app's build by default. For CSS files within the styles directory, you can create multiple stylesheets and use regular CSS APIs like `import` to link them together. If you want to incorporate CSS from an npm package or similar, see [Addons and Dependencies]() for instructions.
+
+## What is `index.html` for?
+
+If HTML markup goes in `hbs` Templates, what is `index.html` for?
+
+The `index.html` file is the entry point for an app. It is not a Template, but rather it is where all the Templates, stylesheets, and JavaScript come together into something the browser can understand.
+
+When you are first getting started in Ember, you will not need to make any changes to `index.html`. There's no need to add any links to other Ember app pages, stylesheets, or scripts in here by hand, since Ember's built-in tools do the work for you.
+
+The most common customizations developers make to `index.html` are when they use a CDN to load assets like fonts or stylesheets. Here's an example:
+
+```html {data-filename=app/index.html}
+  <link integrity="" rel="stylesheet" href="https://my-font-cdn/something.css">
 ```
 
-The `firstName` and `lastName` properties are read from the
-context (the application controller in this case), and rendered inside the
-`<strong>` HTML tag.
+## Understanding a Template's context
 
-To provide a `firstName` and `lastName` to the above template, properties
-must be added to the application controller. If you are following along with
-an Ember CLI application, you may need to create this file:
+A Template only has access to the data it has been given. This is referred to as the Template's "context." For example, to display a property inside a Route's Template, it should be defined in a Controller:
```

Comment:
it's unfortunate we have to make this leap

Reply:
Yeah, I think it'd be better to use a Component as an example here, and then explain why routes are different, are at least address the separation of concerns here somehow